alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -63,33 +101,27 @@ void RunnableObject::start(RunningPlan* rp)
         int64_t wrapperId = (*it)->getId();
 
         BasicPlan* parentPlan = rp->getParent()->getBasicPlan();
-        auto& planAttachment = parentPlan->getPlanAttachment(wrapperId);
-        auto initCall = [this, &planAttachment, parentPlan = parentPlan]() {
-            if (!_blackboard) {
-                _blackboard = std::make_shared<Blackboard>();
-            }
-            _blackboard->impl().clear();
-            if (!planAttachment->setParameters(*parentPlan->getBlackboard(), *_blackboard)) {
-                std::cerr << "Setting parameters failed, supposedly as the context has already changed.  Plan will not be scheduled" << std::endl;
-                return;
-            }
+        const auto keyMapping = parentPlan->getKeyMapping(wrapperId);
+
+        initCall = [this, keyMapping, parentPlan]() {
+            _blackboard = std::make_shared<Blackboard>(_blackboardBlueprint);
+            keyMapping.setInput(parentPlan->getBlackboard().get(), _blackboard.get());
             doInit();
         };
-        _engine->editScheduler().schedule(initCall);
-    } else {
-
-        BasicPlan* parentPlan = rp->getParent() ? rp->getParent()->getBasicPlan() : nullptr;
-        auto initCall = [this, parentPlan = parentPlan]() {
-            // Share Blackboard with parent if we have one, or start fresh otherwise
+    } else if (_inheritBlackboard) {
+        BasicPlan* parentPlan = rp->getParent()->getBasicPlan();
+        initCall = [this, parentPlan]() {
             if (parentPlan) {
                 _blackboard = parentPlan->getBlackboard();
```

Comment:
Does the check for parentPlan do anything here?  case with no parent should now be handled above right?